docusaurus/docs/iOS/guides/working-with-user.md

```diff
@@ -1,3 +1,185 @@
+
+To put the information from above into code, let's say you have an `NetworkManager` which can fetch StreamChat JWT tokens for your user. You'd use a `closure` `TokenProvider` with your StreamChat, so:
+```swift
+let tokenProvider = TokenProvider.closure { chatClient, completion in
+    /// Here, fetch a token locally or use URLSession/Alamofire/etc to fetch
+    /// a token from your backend service and pass it into completion
+    networkManager.fetchToken(for: myUserId) { token, error in
+        if let token = token {
+            completion(.success(token))
+        } else if let error = error {
+            completion(.failure(error))
+        }
+    }
+}
+
+let config = ChatClientConfig(apiKey: <# Your API Key Here #>)
+let chatClient = ChatClient(config: config, tokenProvider: tokenProvider)
+```
+and `ChatClient` will handle the rest, it'll call your closure whenever it needs a valid token. Your closure will be called once per app lifecycle to acquire a valid token (since StreamChat doesn't store JWT tokens) and it'll call the closure whenever the backend reports the token is invalid (ie when expired or revoked).
+
+Since the hosting app "owns" the login/logout logic, whenever a new user logs in, you should call:
+```swift
+chatClient.currentUserController().reloadUserIfNeeded()
+```
+to make sure `ChatClient` calls your `tokenProvider` closure and acquires a new token for the newly logged-in user.
+
+You don't need to "logout" in the common sense. Deallocating `ChatClient` instances for the currently logged-in user means the user will disconnect. If you want to keep your "ChatClient" instance (eg you're using a singleton for it) you can call:
+```swift
+chatClient.connectionController().disconnect()
+```
+
+:::note
+For more information regarding connection & disconnection, please check [Connection Status guide](connection-status).
+:::
+
+If you're using static tokens for developing quick prototypes and want to test login/logout, you can assign a new `tokenProvider` to the `ChatClient` and then call `reloadUserIfNeeded` to "login" using the new token:
+```swift
+chatClient.tokenProvider = .static(newToken)
+chatClient.currentUserController().reloadUserIfNeeded()
+```
+
+## CurrentUser vs User
+
+You can get 2 types of controllers from `ChatClient`: `CurrentUserController` vs `UserController`
+
+### `CurrentUserController`
+
+`CurrentUser` is a wrapper for the currently logged-in user's `User` object. `CurrentUser` includes information about the logged-in user, such as registered devices, muted users, flagged users, and unread count. You interact with your current user via `CurrentUserController`
+
+For all functions available on this controller, please check [CurrentUserController docs](404).
```

Comment:
Don't forget to add the reference docs link after rebase:

`../ReferenceDocs/Sources/StreamChat/Controllers/CurrentUserController/CurrentUserController` (double check if this is right).